### PR TITLE
add tests for Emby.Naming/TV/EpisodePathParser.cs

### PR DESCRIPTION
**Pull request by Narfinger**
This should in the future help to detect working and non working name matchings

**Changes**
Adding tests for Emby.Naming/TV/EpisodePathParser.cs and working tests cases.
There are still a lot of common file names not covered by this test because I prioritized successful tests over tests that are supposed to work.

This should hopefully help in the future to improve the parser.

### CODE REVIEW

**Comment by Narfinger** [comment]
If somebody has some more paths that definitely pass the checks I would be very happy to include them!

**Comment by cvium** [comment]
Did you check the existing tests?
https://github.com/MediaBrowser/Emby.Naming/tree/master/Emby.Naming.Tests

**Comment by Narfinger** [comment]
The commit added a couple of tests from the MediaBrowser.

File: tests/Jellyfin.Naming.Tests/EpisodePathParserTest.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        [Theory]
+        [InlineData("/media/Foo/Foo-S01E01", "Foo", 1, 1)]
+        [InlineData("/media/Foo - S04E011", "Foo", 4, 11)]
+        [InlineData("/media/Foo/Foo s01x01", "Foo", 1, 1)]
+        [InlineData("/media/Foo (2019)/Season 4/Foo (2019).S04E03", "Foo (2019)", 4, 3)]
+        public void ParseEpisodesCorrectly(string path, string name, int season, int episode)
+        {
+            NamingOptions o = new NamingOptions();
+            EpisodePathParser p = new EpisodePathParser(o);
+            var res = p.Parse(path, false);
```
Comment (dkanada) [inline tests/Jellyfin.Naming.Tests/EpisodePathParserTest.cs]: I would use full names for variables to match the rest of the project.

File: tests/Jellyfin.Naming.Tests/EpisodePathParserTest.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            EpisodePathParser p = new EpisodePathParser(o);
+            var res = p.Parse(path, false, null, null, true);
+
+            Assert.True(res.Success);
+            Assert.Equal(name, res.SeriesName);
+            Assert.True(res.SeasonNumber == null);
+            Assert.Equal(episode, res.EpisodeNumber);
+
+            //testing other paths delimeter
+            var res2 = p.Parse(path.Replace("/", "\\"), false, null, null, true);
```
Comment (dkanada) [inline tests/Jellyfin.Naming.Tests/EpisodePathParserTest.cs]: You could probably reuse the first result instead of creating another.

**Comment by JustAMan** [comment]
Tests seem to be duplicated between this PR and #1929, what's the reason for this?

**Comment by Narfinger** [comment]
> Tests seem to be duplicated between this PR and #1929, what's the reason for this?
As mentioned above, #1929 builds on this to include the correct test cases. They can be made independent if necessary.

**Comment by JustAMan** [comment]
So this is essentially blocked by #1929 and should be re-structured when that is merged... I'm going to add "Blocked" label now.

**Pull request merged**